graylog2-server/src/main/java/org/graylog/security/authservice/AuthServiceAuthenticator.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.security.authservice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+public class AuthServiceAuthenticator {
+    private static final Logger LOG = LoggerFactory.getLogger(AuthServiceAuthenticator.class);
+
+    private final GlobalAuthServiceConfig authServiceConfig;
+    private final ProvisionerService provisionerService;
+
+    @Inject
+    public AuthServiceAuthenticator(GlobalAuthServiceConfig authServiceConfig,
+                                    ProvisionerService provisionerService) {
+        this.authServiceConfig = authServiceConfig;
+        this.provisionerService = provisionerService;
+    }
+
+    /**
+     * Tries to authenticate the username with the given password and returns the authenticated username if successful.
+     *
+     * @param authCredentials the authentication credentials
+     * @return the authenticated username
+     */
+    public AuthServiceResult authenticate(AuthServiceCredentials authCredentials) {
+        final Optional<AuthServiceBackend> activeBackend = authServiceConfig.getActiveBackend();
+
+        if (activeBackend.isPresent()) {
+            final AuthServiceResult result = authenticate(authCredentials, activeBackend.get());
+            if (result.isSuccess()) {
+                return result;
+            }
+            // TODO: Do we want the fallback to the default backend here? Maybe it should be configurable?
```

Comment:
My thoughts to this:

- LDAP server logs to graylog
- LDAP server is "broken"
- You want to look at the problem but you can't log in to graylog since LDAP is "broken"
- A fallback user with reader rights (so other than admin) would be helpful

Reply:
@kmerz One fallback user that is always there (at least at the moment) is the root user account. People can also still add users to MongoDB as fallback admin accounts if the external provider breaks.

This comment was more about if we always want the fallback to the MongoDB backend or are there reasons to not do that. To make you use case work, we need the fallback. (and it's also that way at the moment)